hikari/impl/shard.py

```diff
@@ -527,6 +527,12 @@ async def get_user_id(self) -> snowflakes.Snowflake:
             raise RuntimeError("user_id was not known, this is probably a bug")
         return self._user_id
 
+    def _get_ws(self) -> _GatewayTransport:
+        if not self._ws:
+            raise errors.ComponentStateConflictError("Shard is shutting down")
```

Comment:
```suggestion
            raise errors.ComponentStateConflictError("Shard is closed")
```

Reply:
If it was already closed when whatever call triggered this was started then _check_if_alive should've raised before it got this far